README.md

```diff
@@ -1,44 +1,141 @@
 
-# Caliptra Firmware/Software
+# Caliptra firmware and software
 
 This repository contains firmware, libraries, and tools related to the
 [Caliptra Project](https://github.com/chipsalliance/caliptra).
 
-## [rom](/rom/dev)
+## Documentation
 
-"Read Only Memory", the code that is first executed when the chip is powered on.
-Normally the ROM is part of the silicon. 
+- [Caliptra ROM Module Specification](/rom/dev/README.md)
+- [Caliptra FMC Module Specification](/fmc/README.md)
+- [Caliptra Runtime Module Specification](/runtime/README.md)
+- [libcaliptra Guide](/libcaliptra/README.md)
+- [Caliptra GitHub GCP Runner Infrastructure](/ci-tools/github-runner/README.md)
+
+## Directory structure
+
+### [api](/api)
+
+Definitions for the Caliptra mailbox and other basic firmware interfaces.
+
+### [builder](/builder)
+
+Tool for building the Caliptra firmware bundle.
+
+### [cfi](/cfi)
+
+Library containing Control-Flow Integrity attack countermeasure implementations.
+
+### [ci-tools](/ci-tools)
+
+Various tools used for Continuous Integration flows.
+
+### [common](/common)
+
+Common code shared across multiple other code modules.
+
+### [coverage](/coverage)
 
-## [fmc](/fmc/)
+Tool for collecting code coverage metrics from Caliptra tests.
+
+### [cpu](/cpu)
+
+Implementations of CPU-specific features such as NMI and Trap handlers.
+
+### [dpe](/dpe)
+
+DICE Protection Environment submodule (reference to its own repository).
+
+### [drivers](/drivers)
+
+A rust library containing drivers for the Caliptra hardware, intended to be
+used by firmware running on Caliptra's RISC-V cpu.
+
+### [error](/error)
+
+Comprehensive list of all Caliptra error codes.
+
+### [fmc](/fmc/)
 
 "First Mutable Code", the code that the boot ROM measures and jumps to after
 validation succeeds.
 
-## [hw-model](/hw-model/)
+### [hw-latest](/hw-latest/)
+
+Reference to the latest stable Caliptra RTL (reference to its own repository).
+
+### [hw-model](/hw-model/)
 
 A high-level testing library for instantiating and manipulating models of the
 hardware. Intended to target multiple backends, including sw-emulator,
 verilator, and (hopefully) a future FPGA implementation.
 
-## [sw-emulator](/sw-emulator/)
+### [image](/image)
```

Comment:
I need help with a good description for this one

Reply:
"Libraries for generating and verifying Caliptra firmware images"

Reply:
Recommend: Add a period at the end of the description.